apm-sniffer/bytebuddy-patch/src/main/java/net/bytebuddy/agent/builder/SWDescriptionStrategy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package net.bytebuddy.agent.builder;
+
+import net.bytebuddy.agent.builder.AgentBuilder.PoolStrategy.WithTypePoolCache.Simple;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.pool.TypePool;
+import net.bytebuddy.utility.JavaModule;
+import net.bytebuddy.utility.nullability.MaybeNull;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A DescriptionStrategy with interoperable cache provider.
+ */
+public class SWDescriptionStrategy implements AgentBuilder.DescriptionStrategy {
+
+    private Simple typePoolCache = new Simple(new ConcurrentHashMap<>());
+
+    private AgentBuilder.DescriptionStrategy delegate = AgentBuilder.DescriptionStrategy.Default.HYBRID;
+
+    @Override
+    public boolean isLoadedFirst() {
+        return true;
+    }
+
+    @Override
+    public TypeDescription apply(String name,
+                                 @MaybeNull Class<?> type,
+                                 TypePool typePool,
+                                 AgentBuilder.CircularityLock circularityLock,
+                                 @MaybeNull ClassLoader classLoader,
+                                 @MaybeNull JavaModule module) {
+        // use cache first
+        TypePool.CacheProvider cacheProvider = getCacheProvider(classLoader);
+        TypePool.Resolution resolution = cacheProvider.find(name);
+        if (resolution != null && resolution.isResolved()) {
+            return resolution.resolve();
+        }
+        // do as AgentBuilder.DescriptionStrategy.Default.HYBRID
+        return delegate.apply(name, type, typePool, circularityLock, classLoader, module);
+    }
```

Comment:
What is the diff between this and pool first?

Reply:
`POOL_FIRST`  strategy call `typePool.describe(..)` first, simply:
* find from cache
* parse bytecode if not found

But SWDescriptionStrategy is: 
* find from cache
* do as `HYBRID` strategy, create from type class if not null,  then from type pool

```java
HYBRID(true) {
      /** {@inheritDoc} */
      public TypeDescription apply(String name,
                                   @MaybeNull Class<?> type,
                                   TypePool typePool,
                                   CircularityLock circularityLock,
                                   @MaybeNull ClassLoader classLoader,
                                   @MaybeNull JavaModule module) {
          return type == null
                  ? typePool.describe(name).resolve()
                  : TypeDescription.ForLoadedType.of(type);
      }
  },

POOL_FIRST(false) {
        /** {@inheritDoc} */
        public TypeDescription apply(String name,
                                     @MaybeNull Class<?> type,
                                     TypePool typePool,
                                     CircularityLock circularityLock,
                                     @MaybeNull ClassLoader classLoader,
                                     @MaybeNull JavaModule module) {
            TypePool.Resolution resolution = typePool.describe(name);
            return resolution.isResolved() || type == null
                    ? resolution.resolve()
                    : TypeDescription.ForLoadedType.of(type);
        }
    };

```